### PR TITLE
 Allowing overwritting modules to be instantiated during dkg 

### PR DESCRIPTION
**Pull request by ayush-ranjan-official**
Fixes issue: #1956
using `std::env` I passed  some `--extra-module-instance <kind>` in the command line that gets passed as `args: vec<ModuleKind>` and eventually gets chained to the `legacy_init_order_iter` with the initial 3 module instances
which are `["mint", "ln", "wallet"]`.
I have also used `if-else` statements to support multiple user generated modules.
I tried to fix the issue with my best understanding of the problem, do correct me if I am missing something as I am relatively new to rust programming. 
Thanks.

### CODE REVIEW

File: fedimint-core/src/config.rs [hunk 1]
```diff
@@ -321,7 +322,35 @@ impl<M> ModuleGenRegistry<M> {
     where
         M: Clone,
     {
-        for hardcoded_module in ["mint", "ln", "wallet"] {
+        let args: Vec<String> = env::args().collect();
```
Comment (dpc) [inline fedimint-core/src/config.rs:325]: We should parse these via `clap`, and pass here as an argument.

File: fedimint-core/src/config.rs [hunk 2]
```diff
@@ -321,7 +322,35 @@ impl<M> ModuleGenRegistry<M> {
...
+                i += 1; // Move to the next argument.
+            }
+        }
+
+        // Chain the modules instances to the legacy_init_order_iter.
+        let mut legacy_init_order_iter = Vec::new();
+        // fixed 3 module instances
+        let mut initial_module: Vec<String> = ["mint", "ln", "wallet"];
+        //adding all module instances to legacy_init_order_iter.
+        legacy_init_order_iter.append(&mut initial_module);
```
Comment (dpc) [inline fedimint-core/src/config.rs:349]: `.chain(extra_modules)` instead.

File: fedimint-core/src/config.rs (hunk 1 above)
Comment (dpc) [reply fedimint-core/src/config.rs:325]: `fedimint-server` already takes care of passing the args, BTW. We can just stick it there, and then just need to carry it to here.

Comment (ayush-ranjan-official) [reply fedimint-core/src/config.rs:325]: Oh! Let me do that again. Thanks.

**Pull request closed without merging**